lib/server.js

```diff
@@ -844,30 +870,48 @@ WalletService.prototype.createTx = function(opts, cb) {
           return cb(new ClientError('NOTALLOWEDTOCREATETX', 'Cannot create TX proposal during backoff time'));
 
         var copayer = wallet.getCopayer(self.copayerId);
-        var hash = WalletUtils.getProposalHash(opts.toAddress, opts.amount, opts.message, opts.payProUrl);
+        var proposalHeader = Model.TxProposal.create(opts).getHeader();
+        var hash = WalletUtils.getProposalHash.apply(WalletUtils, proposalHeader);
         if (!self._verifySignature(hash, opts.proposalSignature, copayer.requestPubKey))
           return cb(new ClientError('Invalid proposal signature'));
 
-        var toAddress;
-        try {
-          toAddress = new Bitcore.Address(opts.toAddress);
-        } catch (ex) {
+        var outputs = (opts.type == Model.TxProposal.Types.MULTIPLEOUTPUTS)
+          ? opts.outputs
+          : [ { toAddress: opts.toAddress, amount: opts.amount } ];
+        var badAddress = false,
```

Comment:
`_.each` will only break execution when the callback function returns `false`. Since doing something like
```javascript
return cb(new ClientError());
```
does not return `false`, lodash continues with the next cycle.